lib/universalid/contrib/active_record/base_packer.rb

```diff
@@ -150,8 +152,12 @@ def has_many_associations
     def loaded_has_many_relations_by_name
       has_many_associations.each_with_object({}) do |association, memo|
         relation = record.public_send(association.name)
-        next unless relation.loaded?
-        memo[association.name] = relation
+
+        if relation.loaded?
+          memo[association.name] = relation
+        elsif relation.target.any? # new unsaved records
+          memo[association.name] = relation.target
+        end
```

Comment:
We need to append the `relation.target` regardless of the loaded check above.

Reply:
Also, consider renaming `loaded_has_many_relations_by_name` as it doesn't always return a relation.